### PR TITLE
Don't allocate 32K on the stack.

### PR DESCRIPTION
**Pull request by floitsch**
Use malloc to allocate the temporary buffer.

### CODE REVIEW

File: src/os_win.cc [hunk 1]
```diff
@@ -326,11 +331,16 @@ char* OS::getenv(const char* variable) {
   wchar_t* variable_w = to_wide_string(variable);
 
   const int BUFFER_SIZE = 32767;
```
Comment (kasperl) [inline src/os_win.cc:333]: No +1 here?

File: src/os_win.cc [hunk 2]
```diff
@@ -326,11 +331,16 @@ char* OS::getenv(const char* variable) {
...
-  wchar_t buffer[BUFFER_SIZE];
+  wchar_t* buffer = unvoid_cast<wchar_t*>(malloc(BUFFER_SIZE));
   int length_w = GetEnvironmentVariableW(variable_w, buffer, BUFFER_SIZE);
   free(variable_w);
-  if (length_w == 0 || length_w > BUFFER_SIZE) return null;
-  return to_narrow_string(buffer, length_w);
+  if (length_w == 0 || length_w > BUFFER_SIZE) {
```
Comment (kasperl) [inline src/os_win.cc:337]: This is different than the other check.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/os_win.cc (hunk 1 above)
Comment (floitsch) [reply src/os_win.cc:333]: Not needed.
The `GetEnvironmentVariable` function behaves differently...
The `GetModuleFileNameW` function truncates and returns the number of written characters.
The `GetEnvironmentVariableW` function returns the needed length.

File: src/os_win.cc (hunk 2 above)
Comment (floitsch) [reply src/os_win.cc:341]: See the other response.

I added a comment.

**Pull request merged**